applications/OptimizationApplication/custom_utilities/container_variable_data_holder/container_variable_data_holder_base.h

```diff
@@ -0,0 +1,194 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <string>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+// Application includes
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+class HistoricalContainerDataIO
+{
+public:
+    ///@name Public operations
+    ///@{
+
+    template<class TDataType>
+    static TDataType& GetValue(
+        ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable)
+    {
+        return rNode.FastGetSolutionStepValue(rVariable);
+    }
+
+    template<class TDataType>
+    static void SetValue(
+        ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable,
+        const TDataType& rValue)
+    {
+        rNode.FastGetSolutionStepValue(rVariable) = rValue;
+    }
+
+    ///@}
+};
+
+class NonHistoricalContainerDataIO
+{
+public:
+    ///@name Public operations
+    ///@{
+
+    template<class TEntityType, class TDataType>
```

Comment:
I'd change the order of the template parameters to make it more consistent with the historical IO.
```diff
- template<class TEntityType, class TDataType>
+ template<class TDataType, class TEntityType>
```

Reply:
moved to a different file